internal/guest/commonutils/utilities.go

```diff
@@ -24,3 +33,72 @@ func DecodeJSONWithHresult(r io.Reader, v interface{}) error {
 	}
 	return nil
 }
+
+// StartTimeSyncService starts the `chronyd` deamon to keep the UVM time synchronized.  We
+// use a PTP device provided by the hypervisor as a source of correct time (instead of
+// using a network server). We need to create a configuration file that configures chronyd
+// to use the PTP device.  The system can have multiple PTP devices so we identify the
+// correct PTP device by verifying that the `clock_name` of that device is `hyperv`.
+func StartTimeSyncService() error {
```

Comment:
I think it makes more sense for this to just live under `cmd/gcs`. I don't think there's any scenario where something else would want to use this code, and if there is in the future we can move it then.

Reply:
"Utilities" packages are dangerous as they easily become dumping grounds for tons of unrelated functionality.

Reply:
I moved it into `cmd/gcs/main.go`.